src/model_theory/syntax.lean

```diff
@@ -621,17 +622,22 @@ protected def total : L.sentence :=
 
 end relations
 
-section nonempty
+section cardinality
 
 variable (L)
 
-/-- A sentence that indicates a structure is nonempty. -/
-protected def sentence.nonempty : L.sentence := ∃' (&0 =' &0)
+/-- A sentence indicating that a structure has `n` distinct elements. -/
+protected def sentence.card_ge (n) : L.sentence :=
+(((((list.fin_range n).product (list.fin_range n)).filter (λ ij : _ × _, ij.1 ≠ ij.2)).map
+  (λ (ij : _ × _), ∼ ((& ij.1).bd_equal (& ij.2)))).foldr (⊓) ⊤).exs
```

Comment:
I think you can use [`finset.off_diag`](https://leanprover-community.github.io/mathlib_docs/data/finset/prod.html#finset.off_diag) here.

Reply:
Unfortunately, I haven't yet defined these operations for `finset`, because they are only commutative and associative up to semantic equivalence, and there is no `list.off_diag`.

Reply:
Oh, I see, good point.